src/devices/Arduino/ReconnectingNetworkStream.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Arduino
+{
+    internal class ReconnectingNetworkStream : Stream
+    {
+        private readonly Func<Stream> _reconnect;
+        private Stream? _streamImplementation;
+        private object _lock = new object();
+
+        public ReconnectingNetworkStream(Stream? underlyingStream, Func<Stream> reconnect)
+        {
+            _streamImplementation = underlyingStream;
+            _reconnect = reconnect;
+            Connect();
+        }
+
+        public ReconnectingNetworkStream(Func<Stream> connect)
+        : this(null, connect)
+        {
+        }
+
+        public override bool CanRead
+        {
+            get
+            {
+                return SafeExecute(x => x.CanRead, false);
```

Comment:
Nit: all of these can be rewritten to:
`public override bool Foo => SafeExecute(...)`